src/qililab/slurm.py

```diff
@@ -0,0 +1,76 @@
+import logging
+import os
+from types import ModuleType
+
+from IPython.core.magic import needs_local_scope, register_cell_magic
+from IPython.core.magic_arguments import argument, magic_arguments, parse_argstring
+from submitit import AutoExecutor  # pylint: disable=import-error
+
+logger = logging.getLogger()
+num_jobs_to_keep = 10
+
+
+@magic_arguments()
+@argument("-o", "--output", help=("Name of the variables that will be returned from the SLURM job."))
```

Comment:
I happened to me (several times now) that the name of the output doesn't correspond to the name of the variable that I use inside the cell.

What do you think of checking that the name inside `output` corresponds to a variable defined inside the cell?

Here you have a potential implementation of this check:

```python
import ast

def is_variable_used_or_defined(code, variable):
    tree = ast.parse(code)
    for node in ast.walk(tree):
        if isinstance(node, ast.Name) and node.id == variable:
            return True
        if isinstance(node, ast.Assign):
            for target in node.targets:
                if isinstance(target, ast.Name) and target.id == variable:
                    return True
    return False

code = """
results = []

for gain in np.arange(0, 1, 100):
    a = np.array([1, 2, 3])
    results.append(a)
"""

print(is_variable_used_or_defined(code, 'results'))
```

Reply:
The idea is using the function above to check that the `output` name that the user provides is actually used inside the code of the cell.

Reply:
makes sense, yes!

Reply:
Finally, I removed the` if isinstance(node, ast.Name) and node.id == variable` because if the output variable is defined in another cell and any values are assigned to it in the magic cell, submitit considers it is None. 
![image](https://github.com/qilimanjaro-tech/qililab/assets/104149115/a6477465-5355-421e-8a88-9749976f4a6f)

Reply:
Nice catch!!